Userland/Libraries/LibCore/Group.cpp

```diff
@@ -23,7 +23,7 @@ ErrorOr<String> Group::generate_group_file() const
     ScopeGuard grent_guard([] { endgrent(); });
     setgrent();
     errno = 0;
-#ifndef AK_OS_MACOS
+#if !defined(AK_OS_MACOS) && defined(__GLIBC__)
```

Comment:
This isn't quite sufficient. The check for !AK_OS_MACOS implicitly means (AK_OS_LINUX || AK_OS_SERENITY || AK_OS_OPENBSD || AK_OS_FREEBSD || .... etc ). Any checks for glibc should be asking AK_OS_LINUX && glibc.